science/SDRangel/files/leandsdr_dvbs2.h.patch

```diff
@@ -1,8 +1,37 @@
 diff --git a/plugins/channelrx/demoddatv/leansdr/dvbs2.h b/plugins/channelrx/demoddatv/leansdr/dvbs2.h
-index f9dcae651..6179173bc 100644
+index f9dcae651..c71a46b44 100644
 --- plugins/channelrx/demoddatv/leansdr/dvbs2.h
 +++ plugins/channelrx/demoddatv/leansdr/dvbs2.h
-@@ -1091,8 +1091,8 @@ struct s2_frame_receiver : runnable
+@@ -35,6 +35,10 @@
+ #include "ldpc.h"
+ #include "sdr.h"
+ 
++// avoid error: variable length array of non-POD element type 'complex'
++// on old clang; see https://trac.macports.org/ticket/59537
++#define PLSCODES_LENGTH 64
++
+ namespace leansdr
+ {
+ 
+@@ -77,7 +81,7 @@ struct s2_plscodes
+ {
+     // PLS index format MODCOD[4:0]|SHORTFRAME|PILOTS
+     static const int COUNT = 128;
+-    static const int LENGTH = 64;
++    static const int LENGTH = PLSCODES_LENGTH;
+     uint64_t codewords[COUNT];
+     complex<T> symbols[COUNT][LENGTH];
+     s2_plscodes()
+@@ -733,7 +737,7 @@ struct s2_frame_receiver : runnable
+         // Read PLSCODE
+ 
+         uint64_t plscode = 0;
+-        complex<float> pls_symbols[plscodes.LENGTH];
++        complex<float> pls_symbols[PLSCODES_LENGTH];
+         for (int s = 0; s < plscodes.LENGTH; ++s)
```

Comment:
if you are going to make this change, how do you ensure `plscodes.LENGTH` never ever exceeds the hardcoded size `PLSCODES_LENGTH` ?

Reply:
Follow the C++ code chain: `plscodes` is defined below to `s2_plscodes<T> plscodes`; `s2_plscodes<T>` is defined above such that `s2_plscodes<T>::LENGTH = PLSCODES_LENGTH` and is a `static const` type & hence can't be changed at runtime (so, hard-coded of a sort); `PLSCODES_LENGTH` is hard-coded to the value 64, which is the same value as used in the original code (just in a different way). So, for all intensive purposes, the code isn't changed; it's just how that constant value is declared that's changed to make more compiler happy.

Reply:
ok, thanks.